src/lib/FreeC/Pass/ExportPass.hs

```diff
@@ -103,3 +113,23 @@ exportInterface modName = do
   isExported entry = case entryName entry of
     IR.Qual modName' _ -> modName' == modName
     IR.UnQual _        -> False
+
+  -- Qualifies the Coq identifier in the module interface with the module name
+  -- when an entry of the module is exported.
+  -- This ensures that any module that imports the entry uses its qualified
+  -- name, which prevents name conflicts between imported modules.
+  qualifyExportedIdent :: EnvEntry -> EnvEntry
+  qualifyExportedIdent entry
+    | isExported entry = if isConEntry entry
+      then entry { entryIdent      = qualifyIdent (entryIdent entry)
+                 , entrySmartIdent = qualifyIdent (entrySmartIdent entry)
+                 }
+      else entry { entryIdent = qualifyIdent (entryIdent entry) }
+    | otherwise = entry
+
+
+  -- Qualifies an unqualified Coq identifier with the module name.
+  -- A qualified Coq identifer is not modified.
+  qualifyIdent :: Qualid -> Qualid
+  qualifyIdent (         Bare coqName ) = Qualified (Text.pack modName) coqName
```

Comment:
There is a smart constructor [`Coq.ident`](https://github.com/FreeProving/free-compiler/blob/1ca3159363e039f60c2b5cedb10ff5b7938d7f90/src/lib/FreeC/Backend/Coq/Syntax.hs#L68) (which is just an alias for `Text.pack`) for [`Coq.Ident`s and `Coq.ModuleIdent`s](https://github.com/FreeProving/language-coq/blob/c36521c742d0c5a41775c8caf4c514e45d8430c4/src/lib/Language/Coq/Gallina.hs#L90) (which both are just an alias for `Text`). Another alternative would be to add a smart constructor `Coq.qualified :: String -> String -> Coq.Qualid` analogous to [`Coq.bare :: String -> Coq.Qualid`](https://github.com/FreeProving/free-compiler/blob/1ca3159363e039f60c2b5cedb10ff5b7938d7f90/src/lib/FreeC/Backend/Coq/Syntax.hs#L72).

Reply:
A smart constructor would be nicer, but considering `coqName` is already a `Coq.Ident` and not a string, It would be necessary to convert it to a string first in order to have it converted back to a `Coq.Ident` by the smart constructor... so I think it might be better to just use `Coq.ident` here.